src/handlers/generate-erc20-permit.ts

```diff
@@ -74,10 +74,14 @@ export async function generateErc20PermitSignature(
     throw new Error("Provider is not defined");
   }
 
-  const { privateKey } = await decryptKeys(_evmPrivateEncrypted);
-  if (!privateKey) {
-    const errorMessage = "Private key is not defined";
-    _logger.fatal(errorMessage);
+  let privateKey = '';
+  try {
+    const privateKeyDecrypted = await decrypt(_evmPrivateEncrypted, String(process.env.X25519_PRIVATE_KEY));
```

Comment:
Unrelated to your PR:

This confuses me a little because this package isn't a worker or workflow but is being used as an npm package and we import the function to create the permit right? But here we are using `process.env.X25519_PRIVATE_KEY`, I am assuming that this works because the env var is defined in `conversation-rewards`?

If `conversation-rewards` is a core plugin then Ubiquity would always be the one generating permits and not partner's which I'm sure is the intention isn't it?

Reply:
> This confuses me a little because this package isn't a worker or workflow but is being used as an npm package and we import the function to create the permit right?

Yes

> But here we are using `process.env.X25519_PRIVATE_KEY`, I am assuming that this works because the env var is defined in conversation-rewards?

Yes. I suppose that `X25519_PRIVATE_KEY` was left as an "artifact" at the time when `ubiquibot/permit-generation` was meant to be used as a separate plugin.

> If `conversation-rewards` is a core plugin then Ubiquity would always be the one generating permits and not partner's which I'm sure is the intention isn't it?

Ideally partners should generate permits themselves. But since we trade setup complexity for simpler UX (by allowing partners to set their wallet private keys and decrypting them on our side via `X25519_PRIVATE_KEY`) the only way for a partner to generate a permit himself (at least right now) is to:
1. Fork https://github.com/ubiquibot/conversation-rewards
2. Set partner's own `X25519_PRIVATE_KEY`
3. Use a forked version of https://github.com/ubiquibot/conversation-rewards in partner's plugin config

The closest thing we've come so far with the "partners generate permits themselves" is [this](https://github.com/ubiquibot/permit-generation/issues/20#issuecomment-2175340965) issue. In short:
1. Partner sets his wallet private key in github repository secrets in the `ubiquibot-config` repository
2. Kernel calls `ubiquibot-config` as a github action plugin and gets a signed permit

Reply:
> when ubiquibot/permit-generation was meant to be used as a separate plugin.

I'm sure the plan is for it still to be used as a plugin eventually but I understand what you are saying.

> the only way for a partner to generate a permit himself (at least right now) is to:

I don't see why we can't pass it as plugin setting when they install the plugin. 

- If it being leaked is a concern then we can append `_token` to it and GitHub will obscure it for us in logs, hacky but effective. 
- If it's org member liability then really isn't that a partner issue as opposed to our problem? But we could address this I think by adding an additional encryption step that comes from us so that without it `X25519_PRIVATE_KEY` alone cannot be used to decrypt `evmPrivateKeyEncrypted`. We could save this salt in `permit-generation` repo secrets and never expose it to contributors, this way partners can pass their encrypted keys around no problem?
- In order to use the encryption UI you should have an official install and we'd prove that before carrying out the encryption, this makes it harder I guess to reverse engineer things. Fits the idea of the 'setup wizard' also as with auth we can write their config files for them etc.

But I see there are plans to turn the config repo into an action and build more utility into that repo which sounds pretty cool although not a lot of reason for it rn if we can pass these as plugin settings vs plugin env.

Reply:
There's a lot to cover in the post above. I don't fully understand a complete flow for decrypting partner's `evmPrivateKeyEncrypted` you propose, need an example. Perhaps its better to proceed with this discussion in https://github.com/ubiquibot/permit-generation/issues/20.